ci/src/cI_docker.ml

```diff
@@ -0,0 +1,143 @@
+open! Astring
+open Lwt.Infix
+open CI_utils.Infix
+
+module DK = CI_utils.DK
+
+module Image = struct
+  type t = {
+    id : string;
```

Comment:
It would be useful to unify this with https://github.com/avsm/mirage-ci/blob/master/src/docker_build.mli#L14 so that we have one type for Docker images within the CI.  I separated out the sha256 (which is queried after the image is built) from the textual tag, and added a human-readable string as well to describe the purpose of the file (this is probably optional).